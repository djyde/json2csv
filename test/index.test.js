const json2csv = require('../')

describe('json2csv', () => {
  const arr = [
    { name: 'foo', age: 126, isDeleted: false},
    { name: 'bar', age: 122, isDeleted: false},
    { name: 'a', age: 123, isDeleted: true},
    { name: 'b', age: 145, isDeleted: false},
    { name: 'fooc', age: 142, isDeleted: true}
  ]

  it('should match snapshot', () => {
    expect(json2csv(arr, { fields: [ 'name', 'age', 'isDeleted' ] })).toMatchSnapshot()
    expect(json2csv(arr, { fields: [ 'name', 'age' ] })).toMatchSnapshot()
    expect(json2csv(arr, { fields: [ 'name', 'isDeleted', 'age' ] })).toMatchSnapshot()
  })
})