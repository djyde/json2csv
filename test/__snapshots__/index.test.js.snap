// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json2csv should match snapshot 1`] = `
"name,age,isDeleted
foo,126,false
bar,122,false
a,123,true
b,145,false
fooc,142,true"
`;

exports[`json2csv should match snapshot 2`] = `
"name,age
foo,126
bar,122
a,123
b,145
fooc,142"
`;

exports[`json2csv should match snapshot 3`] = `
"name,isDeleted,age
foo,false,126
bar,false,122
a,true,123
b,false,145
fooc,true,142"
`;
